server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -277,8 +375,9 @@ public DiscoveryNode(
      * @param address          the nodes transport address
      * @param attributes       node attributes
      * @param roles            node roles
-     * @param version          the version of the node
+     * @param nodeVersion      the version of the node
      */
+    @Deprecated(forRemoval = true)
```

Comment:
Eventually you will not be able to create a discoverynode without specifying node & transport versions explicitly